libraries/AP_InertialSensor/AP_InertialSensor_Invensensev3.cpp

```diff
@@ -143,24 +145,40 @@ void AP_InertialSensor_Invensensev3::start()
     // initially run the bus at low speed
     dev->set_speed(AP_HAL::Device::SPEED_LOW);
 
-    // always use FIFO
-    fifo_reset();
-
     // grab the used instances
     enum DevTypes devtype;
     switch (inv3_type) {
+    case Invensensev3_Type::IIM42652:
+        devtype = DEVTYPE_INS_IIM42652;
+        fifo_config1 = 0x07;
+        temp_sensitivity = 1.0 / 2.07;
+        break;
     case Invensensev3_Type::ICM42688:
         devtype = DEVTYPE_INS_ICM42688;
+        fifo_config1 = 0x07;
+        temp_sensitivity = 1.0 / 2.07;
         break;
     case Invensensev3_Type::ICM42605:
         devtype = DEVTYPE_INS_ICM42605;
+        fifo_config1 = 0x07;
+        temp_sensitivity = 1.0 / 2.07;
+        break;
+    case Invensensev3_Type::ICM40605:
+        devtype = DEVTYPE_INS_ICM40605;
+        fifo_config1 = 0x0f;
```

Comment:
Nitpick. Makes it easier to read
```suggestion
        fifo_config1 = 0x0F;
```